src/lib/flex/flex/flex.ts

```diff
@@ -196,136 +186,119 @@ export class FlexStyleBuilder extends StyleBuilder {
-  @Input('fxFlex.lt-md') set flexLtMd(val: string) { this._cacheInput('flexLtMd', val); };
-  @Input('fxFlex.lt-lg') set flexLtLg(val: string) { this._cacheInput('flexLtLg', val); };
-  @Input('fxFlex.lt-xl') set flexLtXl(val: string) { this._cacheInput('flexLtXl', val); };
-  /* tslint:enable */
-
-  // Note: Explicitly @SkipSelf on LayoutDirective because we are looking
-  //       for the parent flex container for this flex item.
-  constructor(monitor: MediaMonitor,
-              elRef: ElementRef,
-              @Optional() @SkipSelf() protected _container: LayoutDirective,
-              protected styleUtils: StyleUtils,
-              @Inject(LAYOUT_CONFIG) protected layoutConfig: LayoutConfigOptions,
-              protected styleBuilder: FlexStyleBuilder) {
-    super(monitor, elRef, styleUtils, styleBuilder);
+  protected layoutWatcher?: Subscription;
+
+  protected DIRECTIVE_KEY = 'flex';
+  protected direction = '';
+  protected wrap = false;
+
 
-    this._cacheInput('flex', '');
-    this._cacheInput('shrink', 1);
-    this._cacheInput('grow', 1);
+  @Input('fxShrink')
+  get shrink(): string { return this.flexShrink; }
+  set shrink(value: string) {
+    this.flexShrink = value || '1';
+    this.triggerReflow();
   }
 
-  /**
-   * For @Input changes on the current mq activation property, see onMediaQueryChanges()
-   */
-  ngOnChanges(changes: SimpleChanges) {
-    if (changes['flex'] != null || this._mqActivation) {
-      this._updateStyle();
-    }
+  @Input('fxGrow')
+  get grow(): string { return this.flexGrow; }
+  set grow(value: string) {
+    this.flexGrow = value || '1';
+    this.triggerReflow();
   }
 
-  /**
-   * After the initial onChanges, build an mqActivation object that bridges
-   * mql change events to onMediaQueryChange handlers
-   */
-  ngOnInit() {
-    super.ngOnInit();
-
-    this._listenForMediaQueryChanges('flex', '', (changes: MediaChange) => {
-      this._updateStyle(changes.value);
-    });
-
-    if (this._container) {
-      // If this flex item is inside of a flex container marked with
-      // Subscribe to layout immediate parent direction changes
-      this._layoutWatcher = this._container.layout$.subscribe((layout) => {
-        // `direction` === null if parent container does not have a `fxLayout`
-        this._onLayoutChange(layout);
-      });
+  protected flexGrow = '1';
+  protected flexShrink = '1';
+
+  constructor(protected elRef: ElementRef,
+              protected styleUtils: StyleUtils,
+              @Inject(LAYOUT_CONFIG) protected layoutConfig: LayoutConfigOptions,
+              protected styleBuilder: FlexStyleBuilder,
+              protected marshal: MediaMarshaller) {
+    super(elRef, styleBuilder, styleUtils, marshal);
+    this.marshal.init(this.elRef.nativeElement, this.DIRECTIVE_KEY,
+      this.updateStyle.bind(this));
+    if (this.parentElement) {
+      this.layoutWatcher = this.marshal.trackValue(this.parentElement, 'layout')
```

Comment:
Let's use `untilViewDestroyed` pipe: https://gist.github.com/ThomasBurleson/37653f1f273149e6f6bc07e5b29ea313


```ts
this.marshal
.trackValue(this.parentElement, 'layout')
.pipe( untilViewDestroyed(elRef) )
.subscribe( this.onLayoutChange.bind(this) );
```

Reply:
Just add that gist code to utils.

Reply:
I added similar functionality that I think is less verbose. Let me know what you think

Reply:
Your use of `destroySubject` in the base class is fine IMO. I would add another property however that exposes a single observable 

```ts
protected destroyed$ = this.destroySubject.asObservable()
```